.github/workflows/devdocs.yml

```diff
@@ -0,0 +1,85 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Upload dev docs
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron:  '0 0 * * *'
+
+jobs:
+  devdocs:
+    name: Fetch and upload dev docs
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout Arrow
+        uses: actions/checkout@v2
+        with:
+          repository: apache/arrow
+          path: arrow
+
+      - name: Checkout Crossbow
+        uses: actions/checkout@v2
+        with:
+          repository: ursacomputing/crossbow
+          path: crossbow
+
+      - name: Fetch Crossbow branches
+        run: |
+          cd crossbow
+          git fetch origin +refs/heads/*:refs/remotes/origin/*
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.9'
+          cache: 'pip'
+
+      - name: Install archery
+        run: |
+          pip install -e arrow/dev/archery[crossbow]
+
+      - name: Download the crossbow docs artifact
+        run: |
+          archery crossbow --queue-path ./crossbow download-artifacts --no-fetch --task-filter "*docs" --target-dir downloads nightly-2021-12-14-0
+
+      - uses: actions/checkout@v2
+        with:
+          ref: asf-site
+          path: asf-site
+
+      - name: Add docs
+        run: |
+          cd asf-site
+          echo "$(git log -1)"
+          git config user.name "github-actions[bot]"
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
```

Comment:
```suggestion
          git config user.email "github-actions[bot]@users.noreply.github.com"
```

Reply:
Can we apply this before we merge this?

Reply:
I don't really care about this, but I found online that this is the email address to use if you want to have it appear in github that the "github action" github user is the commit author (otherwise it will just be an unrecognized user icon). 
Now, it's not really important since it's an automated commit anyway, and only influences the icon that is used in the github display, so either way is fine for me.

Reply:
OK.
It seems that both of with/without `41898282+` seem work:

* with `41898282+`: https://github.com/jorisvandenbossche/arrow-site/commits/asf-site
* without `41898282+`: https://github.com/ruby/csv/commits/gh-pages

I prefer simpler one. So I apply this.

Reply:
Ah, if both work, the simpler is of course better (this might have changed more recently). Thanks!